### PR TITLE
BF: fix conversion of bit images to numpy arrays

### PR DESCRIPTION
**Pull request by matthew-brett**
Numpy cannot form arrays from bits.  To convert bit images to numpy,
convert bits to bytes, using numpy.

Fixes gh-350.

### CODE REVIEW

**Comment by homm** [comment]
You do not need numpy to convert bytes to bits, you can do the same with Pillow: `self.tobytes('raw', 'L')`.

**Comment by matthew-brett** [comment]
Ah yes, that's much cleaner - thanks.

File: PIL/Image.py [hunk 1]
```diff
@@ -616,16 +616,21 @@ def _repr_png_(self):
         return b.getvalue()
 
     def __getattr__(self, name):
-        if name == "__array_interface__":
-            # numpy array interface support
-            new = {}
-            shape, typestr = _conv_type_shape(self)
-            new['shape'] = shape
-            new['typestr'] = typestr
+        if name != "__array_interface__":
```
Comment (homm) [inline PIL/Image.py]: Why we can't just create `__array_interface__` property? /cc @wiredfool

Comment (wiredfool) [inline PIL/Image.py]: Sounds like a good idea to me.

**Comment by matthew-brett** [comment]
I converted the previous `__getattr__` logic to an `__array_interface__` property, as suggested.   This is a slight change in API, in that you can no longer set the `__array_interface__` attribute on `Image`.

**Pull request merged**